### PR TITLE
USWDS-Site: Add alert for outdated posts

### LINKED ISSUE #2565: USWDS-Site: Create a component that notifies users of outdated content
## Summary
Create a component that notifies readers of outdated content.

Here is an example of an implementation of something similar from [digital.gov](https://digital.gov/2015/12/31/digitalgovs-2015-year-in-review/):
image

### PR DESCRIPTION
**Pull request by amyleadem**
# Summary

Added an alert component to let users know about outdated content in our "News and updates" posts. 

## Related issue

Closes #2565
Related to PR #2569 

## Preview link

Example "outdated" post: [Working with the U.S. Web Design System team](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-outdated-content-component/whats-new/updates/2021/03/24/working-with-uswds-team/)

> [!Note]
> This component is also applied to all "News and updates" posts from 2017-2021. 

## Problem statement
Our "News and updates" content is several years old, which means that the posts might have broken links or outdated content. We should build a component that notifies users that the content might be outdated.

## Testing and review
1. Confirm the component appropriately notifies users that the content might be outdated
2. Confirm the component is accessible:
    1. Can be found by screen reader users
    2. Sticky positioning does not block content at high zoom 
4. Confirm no spelling or grammatical errors
5. Confirm the component is added to all posts that have an `expiration_date` set in the past (This should include all posts from 2017-2021)

### CODE REVIEW

File: _layouts/post.html [hunk 1]
```diff
@@ -13,6 +13,19 @@
...
 
   <main id="main-content" class="main-content">
+    {% if page.expiration_date %}
+      <!--
+        Note: `%s` converts the date to "seconds since 1970", based on strftime formatting.
+        This allows us to compare dates to see which is greater.
+      -->
+      {% assign today_date = 'now' | date: '%s' %}
+      {% assign expiration_date = page.expiration_date | date: '%s' %}
+      {% assign expired_content = false %}
```
Comment (mahoneycm) [inline _layouts/post.html:23]: Is this line needed? Looks like it was brought over from the `head.html` expired content logic but doesn't seem to add any value here. Let me know if I'm missing anything!

```suggestion

```

**Review by mahoneycm — CHANGES_REQUESTED** [review]
One question about an unused variable. Other than that, looking good 👍

**Review by sarah-sch — APPROVED** [review]
This looks good, @amyleadem. Thank you!

File: _layouts/post.html (hunk 1 above)
Comment (amyleadem) [reply _layouts/post.html]: Good catch! Removed in [42e3340](https://github.com/uswds/uswds-site/pull/2618/commits/42e33405cbe7083825cfca0dbf952686a64ab815)

**Review by mahoneycm — APPROVED** [review]

**Review by annepetersen — APPROVED** [review]
LGTM

**Comment by thisisdano** [comment]
Nice work! I'm updating this PR with the most recent `main` and that should resolve the CI errors

**Review by thisisdano — APPROVED** [review]

**Pull request merged**